app/javascript/readingList/readingList.jsx

```diff
@@ -92,220 +89,153 @@ export class ReadingList extends Component {
-        <a className="readinglist-item" href={item.searchable_reactable_path}>
-          <div className="readinglist-item-title">
-            {item.searchable_reactable_title}
-          </div>
-          <div className="readinglist-item-details">
-            <a
-              className="readinglist-item-user"
-              href={`/${item.reactable_user.username}`}
-            >
-              <img
-                src={item.reactable_user.profile_image_90}
-                alt="Profile Pic"
-              />
-              {item.reactable_user.name}
-・
-              {item.reactable_published_date}
-・
-              {item.reading_time}
-              {' '}
-min read・
-            </a>
-            <span className="readinglist-item-tag-collection">
-              {item.reactable_tags.map(tag => (
-                <a className="readinglist-item-tag" href={`/t/${tag}`}>
-                  #
-                  {tag}
-                </a>
-              ))}
-            </span>
-          </div>
-        </a>
-        <button
-          className="readinglist-archive-butt"
-          onClick={e => this.archive(e, item)}
-          type="button"
-        >
-          {statusView === 'valid' ? 'archive' : 'unarchive'}
-        </button>
-      </div>
-    ));
-    if (readingListItems.length === 0 && itemsLoaded) {
-      if (statusView === 'valid') {
-        allItems = (
-          <div className="readinglist-empty">
-            <h1>
-              {selectedTags.length === 0 && query.length === 0
-                ? 'Your Reading List is Lonely'
-                : 'Nothing with this filter 🤔'}
-            </h1>
-            <h3>
-              Hit the
-              <span className="highlight">SAVE</span>
-              or
-              <span className="highlight">
-                Bookmark
-                <span role="img" aria-label="Bookmark">
-                  🔖
-                </span>
-              </span>
-              to start your Collection
-            </h3>
-          </div>
-        );
-      } else {
-        allItems = (
-          <div className="readinglist-empty">
-            <h1>
-              {selectedTags.length === 0 && query.length === 0
-                ? 'Your Archive List is Lonely'
-                : 'Nothing with this filter 🤔'}
-            </h1>
-          </div>
-        );
-      }
-    }
-    const allTags = availableTags.map(tag => (
-      <a
-        className={`readinglist-tag ${
-          selectedTags.indexOf(tag) > -1 ? 'selected' : ''
-        }`}
-        href={`/t/${tag}`}
-        data-no-instant
-        onClick={e => this.toggleTag(e, tag)}
-      >
-        #
-        {tag}
-      </a>
-    ));
+
+    const isStatusViewValid = this.statusViewValid();
+
+    const archiveButtonLabel = isStatusViewValid ? 'archive' : 'unarchive';
+    const itemsToRender = items.map(item => {
+      return (
+        <ItemListItem item={item}>
+          <ItemListItemArchiveButton
```

Comment:
hooray for child components which Vue calls slots and I guess React calls "containment pattern" 🤷‍♂️